### PR TITLE
Updated the "Getting Started with Laminas" section

### PR DESCRIPTION
**Pull request by visto9259**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

Updated the "Getting Started with Laminas" section of the Tutorial:

- Updated `.gitignore` to ignore `docs/html` instead of `doc/html`
- Updated all screenshots to use the latest Laminas skeleton views
- Fixed a few links and references
- Improved code snippets to use ModuleManager features

### CODE REVIEW

File: docs/book/getting-started/database-and-models.md [hunk 1]
```diff
@@ -208,20 +208,22 @@ To configure the `ServiceManager`, we can either supply the name of the class to
 be instantiated or a factory (closure, callback, or class name of a factory
 class) that instantiates the object when the `ServiceManager` needs it. We start
 by implementing `getServiceConfig()` to provide a factory that creates an
-`AlbumTable`. Add this method to the bottom of the `module/Album/src/Module.php`
+`AlbumTable`. Add the `ServiceProviderInterface` to the class and add this method to the bottom of the `module/Album/src/Module.php`
```
Comment (froschdesign) [inline docs/book/getting-started/database-and-models.md:211]: The entire module class is too complicated, so it should be reduced to the `getConfig` method.
It makes no sense to add the controller to the configuration file `config/module.config.php` first and then remove it again:

- https://docs.laminas.dev/tutorials/getting-started/modules/#configuration
- https://docs.laminas.dev/tutorials/getting-started/database-and-models/#back-to-the-controller

See also [the section "A Typical Module Class"](https://docs.laminas.dev/laminas-modulemanager/module-class/#a-typical-module-class) in the laminas-modulemanager documentation. (the autoloading is no longer needed!)

I would simplify the whole process for the end user and use the [`ReflectionBasedAbstractFactory` of laminas-servicemanager](https://docs.laminas.dev/laminas-servicemanager/reflection-abstract-factory/). At the end, the factories can be created by [command](https://github.com/laminas/laminas-servicemanager/blob/4.0.x/docs/book/v4/ahead-of-time-factories.md) and replace the `ReflectionBasedAbstractFactory`. The headline could be: "Improving Performance".

Comment (froschdesign) [reply docs/book/getting-started/database-and-models.md:211]: It would be good if there is one place for the configuration and you don't have to jump between files.

For the class `Album\Model\AlbumTable` the `ReflectionBasedAbstractFactory` can be used and for the class `Album\Model\AlbumTableGateway` a factory should be created as a class, then the benefits described [here](https://docs.laminas.dev/tutorials/getting-started/database-and-models/#factories ) are used from the beginning.

https://docs.laminas.dev/tutorials/getting-started/database-and-models/#using-servicemanager-to-configure-the-table-gateway-and-inject-into-the-albumtable

Comment (visto9259) [reply docs/book/getting-started/database-and-models.md:211]: I will simplify the config to use module.config.php.
I agree that using factory classes is a cleaner approach.  I use them myself. 
For the newbie, using the `ReflectionBasedAbstractFactory` may be a little hard to follow unless it is explained properly. 
Is the service manager CLI for AOT only a v4 feature? Has the skeleton been updated to use service-manager v4?

Comment (froschdesign) [reply docs/book/getting-started/database-and-models.md:211]: > For the newbie, using the ReflectionBasedAbstractFactory may be a little hard to follow unless it is explained properly.

But the handling is much easier and the explain can be:

_The factory class `ReflectionBasedAbstractFactory` reads the type hints of the constructor and automatically passes the correct services to the constructor. This process is also known as "autowiring"._ 

> Is the service manager CLI for AOT only a v4 feature? Has the skeleton been updated to use service-manager v4?

Not yet, but the "Performance Improvement" section can also be added later.

Comment (visto9259) [reply docs/book/getting-started/database-and-models.md:211]: OK.  I will update accordingly.

Comment (froschdesign) [reply docs/book/getting-started/database-and-models.md:211]: Change as much as you want here, further changes can be made in other pull requests.

Comment (visto9259) [reply docs/book/getting-started/database-and-models.md:211]: I will update the Getting Started section so that it is complete.
The other sections need updating as well and I will do them separately

**Comment by visto9259** [comment]
Updated the PR to simplify factories and service manager configuration.
Also updated the Album Form to use classes for element types

File: docs/book/getting-started/database-and-models.md [hunk 2]
```diff
@@ -199,74 +199,111 @@ of these methods is, hopefully, self-explanatory.
...
+                    ],
+                ],
+            ],
+        ],
+    ],
+    'view_manager' => [
+        'template_path_stack' => [
+            'album' => __DIR__ . '/../view',
+        ],
+    ],
```
Comment (froschdesign) [inline docs/book/getting-started/database-and-models.md:251]: This part was already added in: https://docs.laminas.dev/tutorials/getting-started/routing-and-controllers/#routing-and-controllers

So this can be reduced to:

```suggestion
    'controllers' => [
        // …
    ],
    'router' => [
        // …
    ],
    'view_manager' => [
        // …
    ],
```

File: docs/book/getting-started/database-and-models.md [hunk 3]
```diff
@@ -361,63 +398,53 @@ class AlbumController extends AbstractActionController
...
-use Laminas\ModuleManager\Feature\ConfigProviderInterface;
+````php
+namespace Album\Controller;
+
+use Album\Model\AlbumTable;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+use Psr\Container\ContainerInterface;
 
-class Module implements ConfigProviderInterface
+class AlbumControllerFactory implements FactoryInterface
```
Comment (froschdesign) [inline docs/book/getting-started/database-and-models.md:412]: I would skip this factory because the main goal is to get started quickly and easily. The creation of boilerplate code must be avoided.
We must bear in mind that the getting started tutorial also has an advertising function.

File: docs/book/getting-started/routing-and-controllers.md [hunk 4]
```diff
@@ -151,10 +151,12 @@ until we set up the views. The URLs for each action are:
...
-`http://laminas-mvc-tutorial.localhost/album`        | `Album\Controller\AlbumController::indexAction`
-`http://laminas-mvc-tutorial.localhost/album/add`    | `Album\Controller\AlbumController::addAction`
-`http://laminas-mvc-tutorial.localhost/album/edit`   | `Album\Controller\AlbumController::editAction`
-`http://laminas-mvc-tutorial.localhost/album/delete` | `Album\Controller\AlbumController::deleteAction`
+`http://localhost:8080/album`                        | `Album\Controller\AlbumController::indexAction`
+`http://localhost:8080/album/add`                    | `Album\Controller\AlbumController::addAction`
+`http://localhost:8080/album/edit`                   | `Album\Controller\AlbumController::editAction`
+`http://localhost:8080/album/delete`                 | `Album\Controller\AlbumController::deleteAction`
+
+**Note:** If you are using self-hosted Apache, replace `http://localhost:8080/` by `http://laminas-mvc-tutorial.localhost/`
```
Comment (froschdesign) [inline docs/book/getting-started/routing-and-controllers.md:159]: In this way, some colour can be added:

```suggestion
NOTE: If you are using self-hosted Apache, replace `http://localhost:8080/` by `http://laminas-mvc-tutorial.localhost/`
```
Compare with: https://docs.laminas.dev/laminas-i18n/filters/number-parse/#using-locale

File: docs/book/getting-started/database-and-models.md (hunk 3 above)
Comment (froschdesign) [reply docs/book/getting-started/database-and-models.md:412]: Ah sorry, the `ReflectionBasedAbstractFactory` from above.

Comment (froschdesign) [reply docs/book/getting-started/database-and-models.md:412]: The factory for the class `AlbumTable` is correct.

File: docs/book/getting-started/database-and-models.md (hunk 2 above)
Comment (visto9259) [reply docs/book/getting-started/database-and-models.md]: Done

File: docs/book/getting-started/database-and-models.md (hunk 3 above)
Comment (visto9259) [reply docs/book/getting-started/database-and-models.md]: Done

File: docs/book/getting-started/routing-and-controllers.md (hunk 4 above)
Comment (visto9259) [reply docs/book/getting-started/routing-and-controllers.md]: Done

**Comment by visto9259** [comment]
Updated PR as per review comments.

**Review by froschdesign — APPROVED** [review]

**Comment by froschdesign** [comment]
@visto9259 
Thank you for your time and this contribution! 👍
